env/tictactoe_gym/tictactoe_gym.py

```diff
@@ -12,49 +12,60 @@
-        self.manager: TicTacToeManager = TicTacToeManager(Board(pieces), pieces_to_win)
+    def __init__(self, size: int = 3, pieces_to_win: int = 3, depht: int = 1):
+        self.depht = depht
+        self.size = size
+        self.manager: TicTacToeManager = TicTacToeManager(Board(size), pieces_to_win)
         self.adversary: Adversary = Adversary(self.manager)
-        self.action_space: ActType = ActionSpace(self.manager.find_legal_moves())
+        self.action_space: ActType = ActionSpaceLegalMoves(self.manager.find_legal_moves())
+        self.observation_space = gymnasium.spaces.Box(low=-1, high=1, shape=(1, self.size ** 2), dtype=np.int8)
 
     def reset(self, seed: int | None = None, options: dict[str, Any] | None = None) -> tuple[ObsType, dict[str, Any]]:
         self.manager.reset_board()
         self.action_space.legal_moves = self.manager.find_legal_moves()
+        self.observation_space = gymnasium.spaces.Box(low=-1, high=1, shape=(1, self.size ** 2), dtype=np.int8)
         return self.manager.board.board, {}
 
     def step(self, action: ActType) -> tuple[ObsType, SupportsFloat, bool, bool, dict[str, Any]]:
+        """
+        :param action: Клетка, куда ходит модель
+        :return: (поле, оценка, победа(bool), False, информация о шаге)
+        """
         self.manager.make_move(action)
         reward = self.manager.has_game_ended()
         if reward is not None:
-            return self.manager.board.board, reward, True, False, {}
+            return self.manager.board.board, reward, True, False, {'step': action, 'win': False, 'reward': reward}
         # Умный ход ботяры(глубина выбирается по тому, что вам нужно)
-        self.manager.make_move(self.adversary.search_root(1))
+        self.manager.make_move(self.adversary.search_root(self.depht))
         self.action_space.legal_moves = self.manager.find_legal_moves()
         reward = self.manager.has_game_ended()
-        terminated = True if reward else False
-        return self.manager.board.board, reward, terminated, False, {}
+        terminated = True if reward is not None else False
+        return (self.manager.board.board, reward, terminated, False,
+                {'step': action, 'win': terminated, 'reward': reward})
 
     def render(self, mode='human') -> None:
-        if mode == 'human':
-            self.display()
-
-    def display(self) -> None:
-        for i in range(int(len(self.manager.board.board) ** 0.5)):
-            for j in range(int(len(self.manager.board.board) ** 0.5)):
-                print(f'{self.manager.board.board[i * int(len(self.manager.board.board) ** 0.5) + j]: >2} ', end='')
-            print()
+        match mode:
+            case 'human':
+                display(self.manager.board.board)
+            case _:
+                raise TypeError(f'mode {mode} not supported')
 
     def close(self):
         return 0
 
 
-class ActionSpace(spaces.Space[np.ndarray]):
+class ActionSpaceLegalMoves(spaces.Space):
```

Comment:
Гениальное название.
```suggestion
class MoveSpace(spaces.Space):
```